staging/src/k8s.io/api/bulk/v1alpha1/types.go

```diff
@@ -0,0 +1,206 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +genclient:nonNamespaced
+
+// BulkRequest contains a single request to bulk api.
+type BulkRequest struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// Opaque identifier used to match requests and responses.
+	RequestID string `json:"requestId" protobuf:"bytes,1,name=requestId"`
+
+	// Starts new watch.
+	// +optional
+	Watch *WatchOperation `json:"watch,omitempty" protobuf:"bytes,2,opt,name=watch"`
+
+	// Starts new watch.
+	// +optional
+	WatchList *WatchListOperation `json:"watchList,omitempty" protobuf:"bytes,3,opt,name=watchList"`
+
+	// Stops existing watch.
+	// +optional
+	StopWatch *StopWatchOperation `json:"stopWatch,omitempty" protobuf:"bytes,4,opt,name=stopWatch"`
+
+	// Gets list of items.
+	// +optional
+	List *ListOperation `json:"list,omitempty" protobuf:"bytes,5,opt,name=list"`
+
+	// Gets single item.
+	// +optional
+	Get *GetOperation `json:"get,omitempty" protobuf:"bytes,6,opt,name=get"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +genclient:nonNamespaced
+
+// BulkResponse contains a single response from bulk api.
+type BulkResponse struct {
```

Comment:
Are "responses" always 1:1 with requests?

How about ClientMessage and ServerMessage as names if not?

Reply:
>> Are "responses" always 1:1 with requests?
No. For each request it is exactly 1 response (either success or failure) + some additional responses are not mapped to requests (watch events).

>> How about ClientMessage and ServerMessage as names if not?
Sounds good to me.